### PR TITLE
feat(v9): migrate react-components suite to ship rolluped only dts

### PR DESCRIPTION
**Pull request by Hotell**
Applied yarn workspace-generator migrate-converged-pkg to ship only rolluped type definitions for:

- react-components

From now on, `unstable` api is processed in same fashion like stable - we generate api.md file on any change 🚀

## Related Issue(s)

Fixes partially https://github.com/microsoft/fluentui/issues/22429

### CODE REVIEW

File: packages/react-components/react-components/config/api-extractor.unstable.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
+  "extends": "@fluentui/scripts/api-extractor/api-extractor.common.v-next.json",
+  "mainEntryPointFilePath": "<projectFolder>/dist/types/unstable/index.d.ts",
```
Comment (Hotell) [inline packages/react-components/react-components/config/api-extractor.unstable.json:4]: this is needed to process `unstable` api. thank to this we now have a common approach for `unstable` apis introduction to any package

File: packages/react-components/react-components/etc/react-components.unstable.api.md [hunk 2]
```diff
@@ -0,0 +1,454 @@
+## API Report File for "@fluentui/react-components"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
```
Comment (Hotell) [inline packages/react-components/react-components/etc/react-components.unstable.api.md:3]: new api reported - generated content

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/df7387d8394994e1802cb1354a327b86a4d41604'>df7387d8394994e1802cb1354a327b86a4d41604</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/Avatar-mount_base.html">898</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/Avatar-mount.html">901</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/Button-mount_base.html">558</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/Button-mount.html">554</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProvider-mount_base.html">1801</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProvider-mount.html">1842</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProviderWithTheme-mount_base.html">291</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProviderWithTheme-mount.html">282</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">234</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">268</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">307</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/MakeStyles-mount_base.html">1575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22746/perf-test-react-components/MakeStyles-mount.html">1545</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request merged**